build-logic-settings/README.md

```diff
@@ -0,0 +1,3 @@
+# What is this module about?
+
+This module contains settings plugins applied to our build logic.
```

Comment:
We starting adding a `description` to the sub-project in the build file, for example like here: https://github.com/gradle/gradle/blob/master/subprojects/snapshots/build.gradle.kts#L6

I think we should do the same here, instead of adding an extra readme.